RxExample/RxDataSources/Differentiator/Diff.swift

```diff
@@ -126,14 +126,15 @@ public enum Diff {
     //================================================================================
     // swift dictionary optimizations {
 
-    private struct OptimizedIdentity<E: Hashable> : Hashable {
+    private struct OptimizedIdentity<E: Hashable>: Hashable {
+        func hash(into hasher: inout Hasher) {
+            hasher.combine(identity.pointee)
+        }
 
-        let hashValue: Int
         let identity: UnsafePointer<E>
 
         init(_ identity: UnsafePointer<E>) {
             self.identity = identity
-            self.hashValue = identity.pointee.hashValue
```

Comment:
This was an optimization to memoize `hashValue`. `let hashValue: Int` should be saved regardless and used in ` func hash(into hasher: inout Hasher) {`. This is just a copy of RxDataSources, but still ...

Reply:
I'm gonna give it a different name to avoid the Hashable warning.